src/main/java/meteordevelopment/meteorclient/systems/modules/world/AutoNametag.java

```diff
@@ -45,61 +56,51 @@
             .build()
     );
 
-    private Entity entity;
+
+    private Entity target;
     private boolean offHand;
     private int preSlot;
 
     public AutoNametag() {
         super(Categories.World, "auto-nametag", "Automatically uses nametags on entities without a nametag. WILL nametag ALL entities in the specified distance.");
     }
 
-    @Override
-    public void onDeactivate() {
-        entity = null;
-    }
-
     @EventHandler
     private void onTick(TickEvent.Pre event) {
-        entity = null;
-
-        for (Entity entity : mc.world.getEntities()) {
-            if (!entities.get().getBoolean(entity.getType()) || entity.hasCustomName() || mc.player.distanceTo(entity) > distance.get()) continue;
-
-            boolean findNametag = true;
-            if (mc.player.getInventory().getMainHandStack().getItem() instanceof NameTagItem) {
-                findNametag = false;
-            }
-            else if (mc.player.getInventory().offHand.get(0).getItem() instanceof NameTagItem) {
-                findNametag = false;
-                offHand = true;
-            }
-
-            boolean foundNametag = !findNametag;
-            if (findNametag) {
-                for (int i = 0; i < 9; i++) {
-                    ItemStack itemStack = mc.player.getInventory().getStack(i);
-                    if (itemStack.getItem() instanceof NameTagItem) {
-                        preSlot = mc.player.getInventory().selectedSlot;
-                        InvUtils.swap(i);
-                        foundNametag = true;
-                        break;
-                    }
-                }
-            }
-
-            if (foundNametag) {
-                this.entity = entity;
-
-                if (rotate.get()) Rotations.rotate(Rotations.getYaw(entity), Rotations.getPitch(entity), -100, this::interact);
-                else interact();
-
-                return;
-            }
+        // Nametag in hobar
+        FindItemResult findNametag = InvUtils.findInHotbar(Items.NAME_TAG);
+
+        if (!findNametag.found()) {
+            error("No Nametag in Hotbar");
+            toggle();
+            return;
         }
+
+
+        // Target
+        target = TargetUtils.get(entity -> {
+            if (entity.hasCustomName()) return false;
```

Comment:
maybe compare the custom name against nametag's name, so you could rename already named entities